app/main.py

```diff
@@ -1,13 +1,16 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {"line": error.get("line_number"),
+            "column": error.get("column_number"),
+            "message": error.get("text"), "name": error.get("code"),
+            "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [format_linter_error(error) for error in errors],
+            "path": file_path,
+            "status": "failed" if errors else "passed"}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [format_single_linter_file(file_path, linter_report.get(file_path))
+            for file_path in linter_report]
```

Comment:
```suggestion
    return [
        format_single_linter_file(file_path, linter_report.get(file_path))
        for file_path in linter_report
    ]
```
Please improve readability
Also, iterate through items

Reply:
"iterate through items" - Sorry, I don't understand what is required in this case